addons/payment_ingenico/models/payment.py

```diff
@@ -151,7 +151,17 @@ def filter_key(key):
         items = sorted((k.upper(), v) for k, v in values.items())
         sign = ''.join('%s=%s%s' % (k, v, key) for k, v in items if v and filter_key(k))
         sign = sign.encode("utf-8")
-        shasign = sha1(sign).hexdigest()
+
+        hash_function = self.env['ir.config_parameter'].sudo().get_param('payment_ingenico.hash_function', 'sha1').lower()
```

Comment:
Since we're falling back to `'sha1'` on the next line anyway maybe the default value here is not necessary?

Reply:
Default value is because we call `.lower()` on the result of get_param which would result in traceback if get_param return None